tests/elemental3/generate_image.pm

```diff
@@ -0,0 +1,53 @@
+# Copyright 2023-2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Test Elemental container image
+#   This image is used as a base to build an Elemental container image.
+#   Then, that image will be used to build a Host OS on top, so
+#   it includes the kernel, firmware, bootloader, etc.
+#
+# Maintainer: unified-core@suse.com, ldevulder@suse.com
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+
+use testapi;
+use serial_terminal qw(select_serial_terminal);
+use utils qw(zypper_call);
+
+sub run {
+    select_serial_terminal;
+
+    my $arch = get_required_var('ARCH');
+    my $build = get_required_var('BUILD');
+    my $image = get_required_var('CONTAINER_IMAGE_TO_TEST');
+    my $repo_to_test = get_required_var('REPO_TO_TEST');
+    my $img_filename = "elemental-$build-$arch";
+    my $shared = '/var/shared';
+
+    # Create shared directory
+    assert_script_run("mkdir -p $shared");
+
+    # Add Unified Core repository and install Elemental package
+    assert_script_run("transactional-update --non-interactive run zypper addrepo --check --refresh $repo_to_test elemental");
+    assert_script_run("transactional-update --non-interactive --continue run zypper --gpg-auto-import-keys refresh");
+    assert_script_run("transactional-update --non-interactive --continue pkg install elemental3-toolkit");
+    assert_script_run("transactional-update apply");
```

Comment:
I think you can use here `trup_call` from `lib/transactional.pm`.
```
trup_call("run zypper addrepo --check --refresh $repo_to_test elemental");
trup_call('--continue run zypper --gpg-auto-import-keys refresh');
trup_call('--continue pkg install elemental3-toolkit');
trup_call('apply');
```

Reply:
Good catch. Fixed, thanks!